cert-provider/cert-provider.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+ACME_BIN="$(realpath ~/.acme.sh/acme.sh)"
+
+# colour output helpers...
+reset=$(tput -T xterm sgr0)
+red=$(tput -T xterm setaf 1)
+green=$(tput -T xterm setaf 2)
+yellow=$(tput -T xterm setaf 3)
+blue=$(tput -T xterm setaf 4)
+
+function logError() {
+    echo "${red}[Error]${reset} $1"
+}
+
+function logWarn() {
+    echo "${yellow}[Warn]${reset} $1"
+}
+
+function logInfo() {
+    echo "${blue}[Info]${reset} $1"
+}
+
+function logSuccess() {
+    echo "${green}[Success]${reset} $1"
+}
+
+# join domains with -d separator
+join_domains()  { echo "$1" | sed -e "s/,/ -d /g" -e "s/^/-d /g"; }
+
+DO_CERTS="true"
+if [ -z "$ACTIVE" ]; then
+    logError "ACTIVE must be set. Value can be anything."
+    unset DO_CERTS
+fi
+
+if [ -z "$DOMAINS" ]; then
+    logError "DOMAINS must be set. Value should be a comma-delimited string of domains."
+    unset DO_CERTS
+else
+    ACME_DOMAINS=$(join_domains "$DOMAINS")
```

Comment:
You don't need a subshell/sed here

```suggestion
    IFS=, read -r -a ACME_DOMAINS <<< "$DOMAINS"
    IFS=' ' read -r -a DOMAIN_ARGS <<< "${ACME_DOMAINS[@]/#/-d }"
```